code/modules/reagents/chemistry/recipes/toxins.dm

```diff
@@ -2,56 +2,201 @@
+	thermic_constant = -200
+	H_ion_release = 5.7
+	rate_up_lim = 20
+	purity_min = 0.3
+	reaction_flags = REACTION_PH_VOL_CONSTANT
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_DAMAGING | REACTION_TAG_BRUTE
 
 /datum/chemical_reaction/facid
 	results = list(/datum/reagent/toxin/acid/fluacid = 4)
 	required_reagents = list(/datum/reagent/toxin/acid = 1, /datum/reagent/fluorine = 1, /datum/reagent/hydrogen = 1, /datum/reagent/potassium = 1)
+	mix_message = "The mixture bubbles fiercly."
+	is_cold_recipe = FALSE
 	required_temp = 380
+	optimal_temp = 680
+	overheat_temp = 800
+	optimal_ph_min = 0
+	optimal_ph_max = 2
+	determin_ph_range = 5
+	temp_exponent_factor = 2
+	ph_exponent_factor = 10
+	thermic_constant = -200
+	H_ion_release = -25
+	rate_up_lim = 20
+	purity_min = 0.5
+	reaction_flags = REACTION_PH_VOL_CONSTANT
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_DAMAGING | REACTION_TAG_PLANT | REACTION_TAG_BURN | REACTION_TAG_TOXIN
 
 /datum/chemical_reaction/nitracid
 	results = list(/datum/reagent/toxin/acid/nitracid = 2)
 	required_reagents = list(/datum/reagent/toxin/acid/fluacid = 1, /datum/reagent/nitrogen = 1,  /datum/reagent/hydrogen_peroxide = 1)
+	mix_message = "The mixture bubbles fiercly and gives off a pungent smell."
+	is_cold_recipe = FALSE
 	required_temp = 480
+	optimal_temp = 680
+	overheat_temp = 900
+	optimal_ph_min = 0
+	optimal_ph_max = 4.1
+	determin_ph_range = 5
+	temp_exponent_factor = 2
+	ph_exponent_factor = 10
+	thermic_constant = -200
+	H_ion_release = -20
+	rate_up_lim = 20
+	purity_min = 0.5
+	reaction_flags = REACTION_PH_VOL_CONSTANT
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_DAMAGING | REACTION_TAG_BURN | REACTION_TAG_TOXIN
 
 /datum/chemical_reaction/sulfonal
 	results = list(/datum/reagent/toxin/sulfonal = 3)
 	required_reagents = list(/datum/reagent/acetone = 1, /datum/reagent/diethylamine = 1, /datum/reagent/sulfur = 1)
+	mix_message = "The mixture changes color and becomes clear."
+	is_cold_recipe = FALSE
+	required_temp = 100
+	optimal_temp = 450
+	overheat_temp = 900
+	optimal_ph_min = 4
+	optimal_ph_max = 8
+	determin_ph_range = 2
+	temp_exponent_factor = 1.5
+	ph_exponent_factor = 1.5
+	thermic_constant = 200
+	H_ion_release = 0.13
+	rate_up_lim = 10
+	purity_min = 0.5
+	reaction_flags = REACTION_PH_VOL_CONSTANT
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_DAMAGING | REACTION_TAG_TOXIN | REACTION_TAG_OTHER
 
 /datum/chemical_reaction/lipolicide
 	results = list(/datum/reagent/toxin/lipolicide = 3)
 	required_reagents = list(/datum/reagent/mercury = 1, /datum/reagent/diethylamine = 1, /datum/reagent/medicine/ephedrine = 1)
+	mix_message = "The mixture becomes cloudy."
+	is_cold_recipe = FALSE
+	required_temp = 100
+	optimal_temp = 450
+	overheat_temp = 900
+	optimal_ph_min = 4
+	optimal_ph_max = 8.5
+	determin_ph_range = 2
+	temp_exponent_factor = 1
+	ph_exponent_factor = 0.2
+	thermic_constant = 500
+	H_ion_release = 2.5
+	rate_up_lim = 10
+	purity_min = 0.7
```

Comment:
This (and the other higher values) will convert into the associated failed chem (i.e. Viscous sludge) by when below this value too _at the end of the reaction_. If you don't want this then you'll want to set the results' reagent to have `failed_chem = null`. I think having higher purity requirements for toxins is fine - I just wanted to make sure you knew of this behavour first.

Reply:
Its considered in the reactions and should be no problem, even for non-pH-users 👍